nevergrad/optimization/test_optimizerlib.py

```diff
@@ -137,6 +137,51 @@ def check_optimizer(
 UNSEEDABLE: tp.List[str] = []
 
 
+def buggy_function(x: np.ndarray) -> float:
+    if any(x[::2] > 0.0):
+        return float("nan")
+    if any(x > 0.0):
+        return float("inf")
+    return np.sum(x ** 2)
+
+
+@skip_win_perf  # type: ignore
+@pytest.mark.parametrize("name", registry)  # type: ignore
+def test_infnan(name: str) -> None:
+    optim_cls = registry[name]
+    optim = optim_cls(parametrization=2, budget=70)
+    if not (
+        any(
+            x in name
+            for x in [
+                "EDA",
+                "EMNA",
+                "Stupid",
+                "Large",
+                "TBPSA",
+                "BO",
+                "Noisy",
+                "chain",
+            ]
+        )
+    ):
+        recom = optim.minimize(buggy_function)
+        result = buggy_function(recom.value)
+        if result < 2.0:
+            return
+        assert (  # The "bad" algorithms, most of them originating in CMA's recommendation rule.
+            any(x == name for x in ["WidePSO", "SPSA", "NGOptBase", "Shiwa", "NGO"])
+            or isinstance(optim, optlib.Portfolio)
+            or isinstance(optim, optlib._CMA)
+            or isinstance(optim, optlib._Rescaled)
+            or isinstance(optim, optlib.recaster.SequentialRecastOptimizer)
+            or "NGOpt" in name
+        )  # Second chance!
```

Comment:
if ever this test ends up needing to get more complicated, it would be better to use a for loop and aviod duplicated code (this is how it's done in `test_optimizers`)